test/delta/delta/diff_test.exs

```diff
@@ -189,5 +189,75 @@ defmodule Tests.Delta.Diff do
 
       assert Delta.compose(a, Delta.diff(a, b)) == b
     end
+
+    @tag custom_embeds: [TestEmbed, QuoteEmbed]
+    test "different embeds" do
+      a = [Op.insert(%{"delta" => [Op.insert("hello world")]})]
+      b = [Op.insert(%{"quote" => [Op.insert("goodbye world")]})]
+
+      assert [
+               Op.delete(1),
+               Op.insert(%{"quote" => [Op.insert("goodbye world")]})
+             ] == Delta.diff(a, b)
+
+      assert Delta.compose(a, Delta.diff(a, b)) == b
+    end
+
+    @tag custom_embeds: [QuoteEmbed]
+    test "embeds without handler diff attributes if equal" do
+      a = [Op.insert(%{"quote" => [Op.insert("hello world")]}, %{"author" => "A"})]
+      b = [Op.insert(%{"quote" => [Op.insert("hello world")]}, %{"author" => "B"})]
+
+      assert [
+               Op.retain(1, %{"author" => "B"})
+             ] == Delta.diff(a, b)
+
+      assert Delta.compose(a, Delta.diff(a, b)) == b
+    end
+
+    @tag custom_embeds: [__MODULE__.CaseInsensitiveEmbed]
+    test "embeds consider operations equal" do
+      # Let's imagine the embed is case-insensitive
+      a = [Op.insert(%{"delta" => [Op.insert("HELLO")]})]
+      b = [Op.insert(%{"delta" => [Op.insert("hello")]})]
+
+      assert [] == Delta.diff(a, b)
+
+      # The invariant breaks for obvious reasons
+      refute Delta.compose(a, Delta.diff(a, b)) == b
+    end
+
+    @tag custom_embeds: [__MODULE__.HomographEmbed]
+    test "embeds consider equal inserts as different" do
+      # Let's imagine the embed considers homographs as different words based on
+      # `meaning` attribute
+      a = [Op.insert(%{"delta" => [Op.insert("football")]}, %{"meaning" => "american"})]
+      b = [Op.insert(%{"delta" => [Op.insert("football")]}, %{"meaning" => "soccer"})]
```

Comment:
The main purpose of an op having "insert" and "attributes" keys is that insert is what determines equality and attributes does not. Otherwise we'd just merge/flatten the two. So this not possible and against delta's semantics. 

What is possible is for the embed op's value to mean whatever the embed wants. So this would be okay:

```ex
a = [Op.insert(%{"delta" => [Op.insert("football", %{"meaning" => "american"})]})]
b = [Op.insert(%{"delta" => [Op.insert("football", %{"meaning" => "soccer"})]})]
```

```bash
iex(4)> b = [Op.insert(%{"delta" => [Op.insert("football", %{"meaning" => "soccer"})]})]
[
  %{
    "insert" => %{
      "delta" => [
        %{"attributes" => %{"meaning" => "soccer"}, "insert" => "football"}
      ]
    }
  }
]
```

Here the value corresponding to the map with "delta" as the key is entirely under that embed's discretion. It happens to take the shape of another delta but it bears no meaning. I would avoid doing this in tests to avoid the confusion though (here and all the other added tests).